config/cmake/HDFCompilerFlags.cmake

```diff
@@ -197,6 +197,19 @@ else ()
   endif ()
 endif ()
 
+# Turn off -Winline warning for Developer builds as this flag
+# appears to conflict specifically with the -Og optimization
+# flag and will produce warnings about functions not being
+# considered for inlining under at least GNU compilers
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
```

Comment:
Why not just take it out of the gnu/developer-general file?

Reply:
It still seems like a good warning to have for -O1, -O2, -O3 builds where HDF5_ENABLE_DEV_WARNINGS is on. It might still be able to inform about why a function wasn't inlined and then someone can go inspect. Developer builds have HDF5_ENABLE_DEV_WARNINGS on by default, but if someone is building in RelWithDebInfo or similar with HDF5_ENABLE_DEV_WARNINGS on, then it makes sense to have -Winline there. It's just currently not compatible with what we probably think of as developer builds (developer or debug builds) since we add -Og in both cases.

Reply:
Secondly, it should probably be in the elseif block of line 187, which is where it would get added. Maybe take it out the developer-general file and only add it back in if supported? All in that elseif block.

Reply:
Moving into the elseif makes sense. As for taking the flag out and re-adding it later - that sort of seems to me like it defeats the purpose of having the warning flag files in the first place.

Reply:
It depends on whether it is a usually used or a usually removed choice. Which one is more common?

Reply:
I suppose that depends on how many people usually turn on HDF5_ENABLE_DEV_WARNINGS